app/controllers/DeclarationController.scala

```diff
@@ -63,17 +69,24 @@ class DeclarationController @Inject()(
 
         declaration => {
 
-          // TODO: send registration to backend once able to build instance of EstateRegistration
-
-          for {
-            updatedAnswers <- Future.fromTry(request.userAnswers
-              .set(DeclarationPage, declaration)
-              .flatMap(_.set(SubmissionDatePage, LocalDateTime.now))
-              .flatMap(_.set(TRNPage, "XC TRN 000 000 4911"))
-            )
-            _ <- repository.set(updatedAnswers)
-          } yield {
-            Redirect(controllers.routes.ConfirmationController.onPageLoad())
+          connector.register(declaration) flatMap {
+            case TRNResponse(trn) =>
+              for {
+                updatedAnswers <- Future.fromTry(request.userAnswers
+                  .set(DeclarationPage, declaration)
+                  .flatMap(_.set(SubmissionDatePage, LocalDateTime.now))
+                  .flatMap(_.set(TRNPage, trn))
+                )
+                _ <- repository.set(updatedAnswers)
+              } yield {
+                Redirect(controllers.routes.ConfirmationController.onPageLoad())
+              }
+            case DeclarationResponse.AlreadyRegistered =>
+              Logger.error(s"[DeclarationController] estate already registered")
+              Future.successful(Conflict(errorHandler.badRequestTemplate))
```

Comment:
Not sure what was on the story but I believe there is a design for already registered.

Reply:
@adamconder Nothing in the the ticket from what I can see - could it be in a separate story?